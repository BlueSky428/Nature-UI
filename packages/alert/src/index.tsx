import * as React from 'react';
import clx from 'clsx';
import { FiInfo, FiAlertCircle, FiAlertTriangle, FiCheckCircle } from 'react-icons/fi';

const SUBTLE_TEXT = 'text-gray-800';

export const ALERT_STATUSES = {
  error: {
    bg: 'bg-red-200',
    iconColor: 'text-red-600 mr-3',
    icon: FiAlertCircle,
    variant: {
      solid: 'bg-red-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  info: {
    bg: 'bg-blue-200',
    iconColor: 'text-blue-600 mr-3',
    icon: FiInfo,
    variant: {
      solid: 'bg-blue-600',
      subtle: SUBTLE_TEXT,
    },
  },
  success: {
    bg: 'bg-green-200',
    iconColor: 'text-green-600 mr-3',
    icon: FiCheckCircle,
    variant: {
      solid: 'bg-green-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  warning: {
    bg: 'bg-orange-200',
    iconColor: 'text-orange-600 mr-3',
    icon: FiAlertTriangle,
    variant: {
      solid: 'bg-orange-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
};

interface AlertOptions {
  /**
   * The status of the alert
   */
  status?: keyof typeof ALERT_STATUSES;
  /**
   * The variant of the alert style to use
   */
  variant?: 'solid' | 'subtle';

  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;

  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ElementType;
}

export type AlertProps = AlertOptions;

const BASE_STYLE = 'px-4 py-3 flex items-center';

const Alert: React.ForwardRefExoticComponent<AlertProps> = React.forwardRef(
  (props: AlertProps, ref: React.Ref<any>) => {
    const {
      className = '',
      status = 'success',
      children,
      variant = 'subtle',
      component: Component = 'div',
    } = props;

    // const Component = 'div';

    const VARIANT: string = ALERT_STATUSES[status].variant[variant];

    const componentClass = clx(BASE_STYLE, {
      [className]: className,
      [ALERT_STATUSES[status].bg]: status,
      [VARIANT]: variant,
    });

    const Icon = ALERT_STATUSES[status];
    const iconClasses = clx({
      [VARIANT]: variant,
      'mr-3': variant,
      [Icon.iconColor]: variant !== 'solid',
    });

    const IconComponent = Icon.icon;

    return (
      <Component className={componentClass} ref={ref}>
        {status && <IconComponent className={iconClasses} size={20} />}
        <span>{children ? children : 'This is an alert ðŸ™‚'}</span>
      </Component>
    );
  }
);

Alert.displayName = 'Alert';

export default Alert;
