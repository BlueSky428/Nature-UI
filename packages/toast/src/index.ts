export * from './create-standalone-toast';
export * from './toast';
export * from './toast.placement';
export * from './toast.provider';
export * from './toast.types';
export * from './use-toast';
