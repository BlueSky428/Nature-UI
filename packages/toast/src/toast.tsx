import {
  Alert,
  AlertDescription,
  AlertIcon,
  AlertProps,
  AlertTitle,
} from '@nature-ui/alert';
import { CloseButton } from '@nature-ui/close-button';
import { nature } from '@nature-ui/system';
import { isFunction } from '@nature-ui/utils';
import * as React from 'react';
import type { RenderProps } from './toast.types';
import type { UseToastOptions } from './use-toast';

export interface ToastProps
  extends UseToastOptions,
    Omit<AlertProps, keyof UseToastOptions> {
  onClose?: () => void;
}

export const Toast: React.FC<ToastProps> = (props) => {
  const { status, variant, id, title, isClosable, onClose, description } =
    props;

  const alertTitleId =
    typeof id !== 'undefined' ? `toast-${id}-title` : undefined;

  return (
    <Alert
      status={status}
      variant={variant}
      id={String(id)}
      alignItems='start'
      borderRadius='md'
      boxShadow='lg'
      paddingEnd={8}
      textAlign='start'
      width='auto'
      aria-labelledby={alertTitleId}
    >
      <AlertIcon />
      <nature.div flex='1' maxWidth='100%'>
        {title && <AlertTitle id={alertTitleId}>{title}</AlertTitle>}
        {description && (
          <AlertDescription display='block'>{description}</AlertDescription>
        )}
      </nature.div>
      {isClosable && (
        <CloseButton
          size='sm'
          onClick={onClose}
          position='absolute'
          insetEnd={1}
          top={1}
        />
      )}
    </Alert>
  );
};

export function createRenderToast(
  options: UseToastOptions & {
    toastComponent?: React.FC<ToastProps>;
  } = {},
) {
  const { render, toastComponent: ToastComponent = Toast } = options;
  const renderToast: React.FC<RenderProps> = (props) => {
    if (isFunction(render)) {
      return render(props) as JSX.Element;
    }
    return <ToastComponent {...props} {...options} />;
  };
  return renderToast;
}
