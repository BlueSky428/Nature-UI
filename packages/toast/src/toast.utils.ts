import { objectKeys } from '@nature-ui/utils';
import {
  ToastPosition,
  ToastOptions,
  ToastState,
  ToastId,
} from './toast.types';

/**
 * Given an array of toasts for a specific position.
 * It returns the toast that matches the `id` passed
 */
export const findById = (arr: ToastOptions[], id: ToastId) => {
  return arr.find((toast) => toast.id === id);
};

/**
 * Given the toast manager state, finds the position of the toast that
 * matches the `id`
 */
export const getToastPosition = (toasts: ToastState, id: ToastId) => {
  let position: ToastPosition | undefined;

  objectKeys(toasts).forEach((pos) => {
    const found = findById(toasts[pos], id);
    if (found) position = pos;
  });

  return position;
};

/**
 * Given the toast manager state, finds the toast that matches
 * the id and return it's position and index
 */
export const findToast = (toasts: ToastState, id: ToastId) => {
  const position = getToastPosition(toasts, id);

  const index = position
    ? toasts[position].findIndex((toast) => String(toast.id) === String(id))
    : -1;

  return {
    position,
    index,
  };
};

/**
 * Given the toast manager state, checks if a specific toast is
 * still in the state, which means it's still visible on screen.
 */
export const isVisible = (toasts: ToastState, id: ToastId) => {
  let found: any;

  Object.values(toasts).forEach((_toasts) => {
    found = _toasts.find((toast) => toast.id === id);
  });

  return !!found;
};

/**
 * Get's the styles to be applied to a toast's container
 * based on it's position in the manager
 */
export const getToastStyle = (position: ToastPosition) => {
  const style: React.CSSProperties = {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  };

  if (position.includes('right')) {
    style.alignItems = 'flex-end';
  } else if (position.includes('left')) {
    style.alignItems = 'flex-start';
  }
  return style;
};
