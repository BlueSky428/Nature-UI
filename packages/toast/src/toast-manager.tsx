import { Interpolation, Theme } from '@emotion/react';
import { nature } from '@nature-ui/system';
import { objectKeys } from '@nature-ui/utils';
import { AnimatePresence } from 'framer-motion';
import * as React from 'react';
import { Toast } from './toast';
import {
  CloseAllToastsOptions,
  ToastId,
  ToastMessage,
  ToastOptions,
  ToastPosition,
  ToastState,
} from './toast.types';
import { findToast, getToastPosition } from './toast.utils';

export interface Methods {
  notify: (message: ToastMessage, options: CreateToastOptions) => ToastId;
  closeAll: (options?: CloseAllToastsOptions) => void;
  close: (id: ToastId) => void;
  update: (id: ToastId, options: CreateToastOptions) => void;
  isActive: (id: ToastId) => boolean;
}

interface Props {
  notify: (methods: Methods) => void;
}

type State = { [K in ToastPosition]: ToastOptions[] };

type CreateToastOptions = Partial<
  Pick<
    ToastOptions,
    | 'status'
    | 'duration'
    | 'position'
    | 'id'
    | 'onCloseComplete'
    | 'containerStyle'
  >
>;

/**
 * Manages the creation and removal of toasts
 * across all corners ("top", "bottom", etc)
 */
export class ToastManager extends React.Component<Props, State> {
  /**
   * Static id counter to create unique ids
   * for each toast
   */
  static counter = 0;

  state: ToastState = {
    top: [],
    'top-left': [],
    'top-right': [],
    'bottom-left': [],
    bottom: [],
    'bottom-right': [],
  };

  constructor(props: Props) {
    super(props);

    /**
     * State to track all the toast across all positions
     */
    const methods = {
      notify: this.notify,
      closeAll: this.closeAll,
      close: this.closeToast,
      update: this.updateToast,
      isActive: this.isVisible,
    };

    props.notify(methods);
  }

  /**
   * Function to actually create a toast and add it
   * to state at the specified position
   */
  notify = (message: ToastMessage, options: CreateToastOptions) => {
    const toast = this.createToast(message, options);
    const { position, id } = toast;

    this.setState((prevToasts) => {
      /**
       * - If the toast is positioned at the top edges, the
       * recent toast stacks on top of the other toasts.
       *
       * - If the toast is positioned at the bottom edges, the recent
       * toast stacks below the other toasts.
       */
      const isTop = position.includes('top');

      return {
        ...prevToasts,
        [position]: isTop
          ? [toast, ...prevToasts[position]]
          : [...prevToasts[position], toast],
      };
    });

    return id;
  };

  /**
   * Update a specific toast with new options based on the
   * passed `id`
   */
  updateToast = (id: ToastId, options: CreateToastOptions) => {
    this.setState((prevState) => {
      const nextState = { ...prevState };
      const { position, index } = findToast(nextState, id);

      if (position && index !== -1) {
        nextState[position][index] = {
          ...nextState[position][index],
          ...options,
        };
      }

      return nextState;
    });
  };

  /**
   * Close all toasts at once
   */
  closeAll = ({ positions }: CloseAllToastsOptions = {}) => {
    this.setState((prevState) => {
      const allPositions: ToastPosition[] = [
        'top',
        'bottom',
        'top-left',
        'top-right',
        'bottom-left',
        'bottom-right',
      ];

      const positionsToClose = positions ?? allPositions;

      return positionsToClose.reduce((acc, position) => {
        acc[position] = prevState[position].map((toast) => ({
          ...toast,
          requestClose: true,
        }));

        return acc;
      }, {});
    });
  };

  /**
   * Create properties for a new toast
   */
  createToast = (message: ToastMessage, options: CreateToastOptions) => {
    const id = options.id ?? ++ToastManager.counter;

    const position = options.position ?? 'top';

    return {
      id,
      message,
      position,
      duration: options.duration,
      onCloseComplete: options.onCloseComplete,
      onRequestRemove: () => this.removeToast(String(id), position),
      status: options.status,
      requestClose: false,
      containerStyle: options.containerStyle,
    };
  };

  /**
   * Requests to close a toast based on it's id and position
   */
  closeToast = (id: ToastId) => {
    this.setState((prevState) => {
      const position = getToastPosition(prevState, id);

      if (!position) return prevState;

      return {
        ...prevState,
        [position]: prevState[position].map((toast) => {
          if (toast.id === id) {
            return {
              ...toast,
              requestClose: true,
            };
          }
          return toast;
        }),
      };
    });
  };

  /**
   * Delete a toast record at it's position
   */
  removeToast = (id: ToastId, position: ToastPosition) => {
    this.setState((prevState) => {
      return {
        ...prevState,
        [position]: prevState[position].filter(
          (toast) => String(toast.id) !== String(id),
        ),
      };
    });
  };

  isVisible = (id: ToastId) => {
    const { position } = findToast(this.state, id);
    return Boolean(position);
  };

  /**
   * Compute the style of a toast based on it's position
   */
  getStyle = (position: ToastPosition): Interpolation<Theme> => {
    const isTopOrBottom = position === 'top' || position === 'bottom';
    const margin = isTopOrBottom ? '0 auto' : undefined;

    const top = position.includes('top')
      ? 'env(safe-area-inset-top, 0px)'
      : undefined;
    const bottom = position.includes('bottom')
      ? 'env(safe-area-inset-bottom, 0px)'
      : undefined;
    const right = !position.includes('left')
      ? 'env(safe-area-inset-right, 0px)'
      : undefined;
    const left = !position.includes('right')
      ? 'env(safe-area-inset-left, 0px)'
      : undefined;

    return {
      position: 'fixed',
      zIndex: 5500,
      pointerEvents: 'none',
      display: 'flex',
      flexDirection: 'column',
      margin,
      top,
      bottom,
      right,
      left,
    };
  };

  render() {
    return objectKeys(this.state).map((position: ToastPosition) => {
      const { state } = this;
      const toasts = state[position];
      return (
        <nature.ul
          key={position}
          id={`nature-toast-manager-${position}`}
          css={this.getStyle(position)}
        >
          <AnimatePresence initial={false}>
            {toasts.map((toast) => (
              <Toast key={toast.id} {...toast} />
            ))}
          </AnimatePresence>
        </nature.ul>
      );
    });
  }
}
