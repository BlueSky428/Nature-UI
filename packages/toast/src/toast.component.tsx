import { useTimeout, useUpdateEffect } from '@nature-ui/hooks';
import { nature } from '@nature-ui/system';
import { runIfFn, __DEV__ } from '@nature-ui/utils';
import { motion, useIsPresent, Variants } from 'framer-motion';
import React from 'react';
import { ToastProviderProps } from './toast.provider';
import type { ToastOptions } from './toast.types';
import { getToastStyle } from './toast.utils';

const toastMotionVariants: Variants = {
  initial: (props) => {
    const { position } = props;

    const dir = ['top', 'bottom'].includes(position) ? 'y' : 'x';

    let factor = ['top-right', 'bottom-right'].includes(position) ? 1 : -1;
    if (position === 'bottom') factor = 1;

    return {
      opacity: 0,
      [dir]: factor * 24,
    };
  },
  animate: {
    opacity: 1,
    y: 0,
    x: 0,
    scale: 1,
    transition: {
      duration: 0.4,
      ease: [0.4, 0, 0.2, 1],
    },
  },
  exit: {
    opacity: 0,
    scale: 0.85,
    transition: {
      duration: 0.2,
      ease: [0.4, 0, 1, 1],
    },
  },
};

export interface ToastComponentProps
  extends ToastOptions,
    Pick<ToastProviderProps, 'motionVariants' | 'toastSpacing'> {}

export const ToastComponent = React.memo((props: ToastComponentProps) => {
  const {
    id,
    message,
    onCloseComplete,
    onRequestRemove,
    requestClose = false,
    position = 'bottom',
    duration = 5000,
    containerStyle,
    motionVariants = toastMotionVariants,
    toastSpacing = '0.5rem',
  } = props;

  const [delay, setDelay] = React.useState(duration);
  const isPresent = useIsPresent();

  useUpdateEffect(() => {
    if (!isPresent) {
      onCloseComplete?.();
    }
  }, [isPresent]);

  useUpdateEffect(() => {
    setDelay(duration);
  }, [duration]);

  const onMouseEnter = () => setDelay(null);
  const onMouseLeave = () => setDelay(duration);

  const close = () => {
    if (isPresent) onRequestRemove();
  };

  React.useEffect(() => {
    if (isPresent && requestClose) {
      onRequestRemove();
    }
  }, [isPresent, requestClose, onRequestRemove]);

  useTimeout(close, delay);

  const containerStyles = React.useMemo(
    () => ({
      pointerEvents: 'auto',
      maxWidth: 560,
      minWidth: 300,
      margin: toastSpacing,
      ...containerStyle,
    }),
    [containerStyle, toastSpacing],
  );

  const toastStyle = React.useMemo(() => getToastStyle(position), [position]);

  return (
    <motion.li
      layout
      className='nature-toast'
      variants={motionVariants}
      initial='initial'
      animate='animate'
      exit='exit'
      onHoverStart={onMouseEnter}
      onHoverEnd={onMouseLeave}
      custom={{ position }}
      style={toastStyle}
    >
      <nature.div
        role='status'
        aria-atomic='true'
        className='nature-toast__inner'
        css={containerStyles as any}
      >
        {runIfFn(message, { id, onClose: close })}
      </nature.div>
    </motion.li>
  );
});

if (__DEV__) {
  ToastComponent.displayName = 'ToastComponent';
}
