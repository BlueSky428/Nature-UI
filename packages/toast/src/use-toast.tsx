import * as React from 'react';
import {
  AlertDescription,
  AlertIcon,
  AlertTitle,
  AlertWrapper,
  ALERT_STATUSES,
} from '@nature-ui/alert';
import { CloseButton } from '@nature-ui/close-button';
import { clsx, nature } from '@nature-ui/system';
import { isFunction, merge } from '@nature-ui/utils';

import { toast } from './toast.class';
import { RenderProps, ToastOptions, ToastId } from './toast.types';

export interface UseToastOptions {
  /**
   * The placement of the toast
   *
   * @default "bottom"
   */
  position?: ToastOptions['position'];
  /**
   * The delay before the toast hides (in milliseconds)
   * If set to `null`, toast will never dismiss.
   *
   * @default 5000 ( = 5000ms )
   */
  duration?: ToastOptions['duration'];
  /**
   * Render a component toast component.
   * Any component passed will receive 2 props: `id` and `onClose`.
   */
  render?(props: RenderProps): React.ReactNode;
  /**
   * The title of the toast
   */
  title?: string;
  /**
   * The description of the toast
   */
  description?: string;
  /**
   * If `true`, toast will show a close button
   */
  isCloseable?: boolean;
  /**
   * The alert component `variant` to use
   */
  variant?: string;
  /**
   * The status of the toast.
   */
  status?: keyof typeof ALERT_STATUSES;
  /**
   * The `id` of the toast.
   *
   * Mostly used when you need to prevent duplicate.
   * By default, we generate a unique `id` for each toast
   */
  id?: ToastId;
  /**
   * Callback function to run side effects after the toast has closed.
   */
  onCloseComplete?: () => void;
}

export type IToast = UseToastOptions;

const DivTag = nature('div');
const Toast = (props: any) => {
  const {
    status = 'success',
    variant,
    id,
    title,
    isCloseable,
    onClose,
    description,
    className = '',
    ...rest
  } = props;

  return (
    <AlertWrapper
      status={status}
      variant={variant}
      id={id}
      className={clsx(className, 'text-left shadow-lg rounded-md m-2 p-4')}
      style={{
        alignItems: 'start',
      }}
      {...rest}
    >
      <AlertIcon />
      <DivTag className='flex-1'>
        {title && <AlertTitle>{title}</AlertTitle>}
        {description && (
          <AlertDescription marginTop='px' lineHeight='short'>
            {description}
          </AlertDescription>
        )}
      </DivTag>
      {isCloseable && (
        <CloseButton
          size='sm'
          onClick={onClose}
          className='absolute right-0 top-0 mr-2 mt-2'
        />
      )}
    </AlertWrapper>
  );
};

const defaults = {
  duration: 5000,
  position: 'bottom',
  variant: 'solid',
} as const;

/**
 * React hook used to create a function that can be used
 * to show toasts in an application.
 */
export const useToast = () => {
  const toastImpl = (options: UseToastOptions) => {
    const { render } = options;

    const Message = (props: RenderProps) => (
      <>
        {isFunction(render) ? (
          render(props)
        ) : (
          <Toast
            {...{
              ...props,
              ...opts,
            }}
          />
        )}
      </>
    );
    const opts = merge(defaults, options);

    return toast.notify(Message, opts);
  };

  toastImpl.close = toast.close;
  toastImpl.closeAll = toast.closeAll;

  toastImpl.update = (id: ToastId, options: Omit<UseToastOptions, 'id'>) => {
    const { render, ...rest } = options;

    const opts = merge(defaults, rest) as any;

    toast.update(id, {
      ...opts,
      // eslint-disable-next-line react/display-name
      message: (props) => (
        <>
          {isFunction(render) ? (
            render(props)
          ) : (
            <Toast
              {...{
                ...props,
                ...opts,
              }}
            />
          )}
        </>
      ),
    });
  };

  toastImpl.isActive = toast.isActive;

  return toastImpl;
};

export default useToast;
