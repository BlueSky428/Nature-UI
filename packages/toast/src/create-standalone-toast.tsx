import { noop } from '@nature-ui/utils';
import * as React from 'react';
import { createRenderToast } from './toast';
import { getToastPlacement } from './toast.placement';
import {
  CreateToastOptions,
  ToastMethods,
  ToastProvider,
  ToastProviderProps,
} from './toast.provider';
import { ToastId, ToastMessage } from './toast.types';
import { UseToastOptions } from './use-toast';

const defaults: UseToastOptions = {
  duration: 5000,
  position: 'bottom',
  variant: 'solid',
};

export interface CreateStandAloneToastParam
  extends Omit<ToastProviderProps, 'children'> {
  defaultOptions: UseToastOptions;
}

export interface CallableToastMethods extends ToastMethods {
  (message: ToastMessage, options: CreateToastOptions): void;
}
export const defaultStandaloneParam: CreateStandAloneToastParam &
  Required<Omit<CreateStandAloneToastParam, keyof ToastProviderProps>> = {
  defaultOptions: defaults,
};

export function createStandaloneToast({
  defaultOptions = defaultStandaloneParam.defaultOptions,
  motionVariants,
  toastSpacing,
  component,
}: CreateStandAloneToastParam = defaultStandaloneParam) {
  const ref = React.createRef<CallableToastMethods>();

  const ToastContainer = () => (
    <ToastProvider
      ref={ref}
      defaultOptions={defaultOptions}
      motionVariants={motionVariants}
      toastSpacing={toastSpacing}
      component={component}
    />
  );

  const normalizeToastOptions = (options?: UseToastOptions) => ({
    ...defaultOptions,
    ...options,
    position: getToastPlacement(options?.position, 'ltr'),
  });

  const toast = (options?: UseToastOptions) => {
    const normalizedToastOptions = normalizeToastOptions(options);
    const Message = createRenderToast(normalizedToastOptions);

    return ref.current?.notify(Message, normalizedToastOptions);
  };

  toast.update = (id: ToastId, options: Omit<UseToastOptions, 'id'>) => {
    if (!id) return;

    const normalizedToastOptions = normalizeToastOptions(options);
    const Message = createRenderToast(normalizedToastOptions);

    ref.current?.update(id, {
      ...normalizedToastOptions,
      message: Message,
    });
  };

  toast.notify = ref.current?.notify ?? noop;
  toast.closeAll = ref.current?.closeAll ?? noop;
  toast.close = ref.current?.close ?? noop;
  toast.isActive = ref.current?.isActive ?? noop;

  return {
    ToastContainer,
    toast,
  };
}
