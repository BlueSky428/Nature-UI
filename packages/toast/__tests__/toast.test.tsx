import * as React from 'react';
import { render } from '@nature-ui/test-utils';
import { Button } from '@nature-ui/button';

import { useToast } from '../src';

test('Spinner renders correctly', () => {
  const Component = () => {
    const toast = useToast();

    return (
      <Button
        onClick={() =>
          toast({
            title: 'Account created.',
            description: "We've created your account for you.",
            status: 'success',
            duration: 9000,
            isCloseable: true,
          })
        }
      >
        Show Success Toast
      </Button>
    );
  };

  const { asFragment } = render(<Component />);

  expect(asFragment()).toMatchSnapshot();
});
