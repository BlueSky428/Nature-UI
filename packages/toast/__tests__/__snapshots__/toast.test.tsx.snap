// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<DocumentFragment>
  <button
    class="px-4 focus:shadow-outline focus:outline-none rounded font-semibold relative overflow-hidden align-middle inline-flex justify-center items-center leading-normal bg-gray-200 text-gray-700 hover:bg-gray-300 css-mqia37 css-1x86u9b css-14d0z92"
  >
    Show Success Toast
  </button>
</DocumentFragment>
`;
